### ISSUE TITLE
npm.bootstrap returns error, even though it eventually succeeds

### ISSUE THREAD

**Issue by c10b10** [saltstack/salt#24045]
I've been running a salt sls using a salt-ssh and `npm.bootstrap` always errors out with:

```sh
Error Bootstrapping '/var/www/...':
```

If I check on the server, I see that the `npm.installed` succeeded.

I tried debugging the issue by editing the `npm.py` module and state files, but it seems that salt-ssh acts as if I haven't changed anything (Is there some kind of cache active?)

I'm proposing that this might be a bug.

Tried this on both v2014.7.1_1 and v2015.5.0.

**Comment by jfindlay** [comment]
@c10b10, is it possible for you to provide the debug log output, the command(s) and npm state(s) you're using?

**Comment by eliasp** [comment]
Fixed in https://github.com/saltstack/salt/commit/27d9c0f8fc61a9d0ec84bed658f0321e910ffa19

**Comment by jfindlay** [comment]
@eliasp, I do not see how removing the `else` clause in the exec module function would prevent the `Error bootstrapping` message.  Can you explain your reasoning?  Thanks.

**Comment by eliasp** [comment]
The state function `npm.bootstrap` sets `pkg=None` as argument when calling the exeuction module function `npm.install`.
This had caused the removed `else` block to kick in, which in turn then returned the string `No package name specified` to the state function.

The state function does:
```Python
288     if isinstance(call, str):
289         ret['result'] = False
290         ret['comment'] = 'Could not bootstrap directory'
291     else:
292         ret['result'] = True
293         ret['changes'] = {name: 'Bootstrapped'}
294         ret['comment'] = 'Directory was successfully bootstrapped'
```
…which means: whatever string the execution module will return, the state will fail.

**Comment by jfindlay** [comment]
This is fixed in #27029, which is in 2015.8.0 and will come out in 2015.5.6.

**Comment by jfindlay** [comment]
@eliasp, thanks for the extra explanation.
